app/models/user.rb

```diff
@@ -1,4 +1,11 @@
 class User < ApplicationRecord
+  enum role: %i[user admin]
+  after_initialize :set_default_role, if: :new_record?
+
+  def set_default_role
+    self.role ||= :user
```

Comment:
I prefer encapsulating creation logic in a service object over using ActiveRecord callbacks, as the callback route has bitten me in the past, but its fine to go this route for now as its only one thing we piece of logic we need to add at the time of record creation.